src/error_handler.ts

```diff
@@ -1,6 +1,25 @@
 
-const error_handler = ( request: Request ): Response => { 
-  return new Response('Not Found.', { status: 404 })
+// Handle the CaptureStackTrace from NodeJS
+// Add a logger as well with leveled logging
+export class ErrorHandler extends Error {
+	public readonly statusCode: number = 0;
+	public readonly method: string = '';
+
+	constructor(msg: any, method?: string, statusCode?: number) {
+		super()
+		this.message = msg
+		if (method) this.method = method
+		if (statusCode) this.statusCode = statusCode
+	}
+
+	// use generic maybe? or try method overriding with declaration merging
+	static throw(opts: { msg: string, method?: string, statusCode?: number, headers?: HeadersInit }): Response {
```

Comment:
@jay-dee7 This method should be responsible of throwing server-side `Error` instances or make responses.

I suggest the following approach:

- Add a `makeResponse` method.
- Wrap `throw` under `throwOrRespond`.
- Add generics to `throwOrRespond`.
- Add param validation to `throwOrRespond`.
- Handle execution based on the validation outcome through either `throw` or `makeResponse`.

Reply:
@Eengineer1 can we do this in another PR? I'd also like to tackle the logging stuff

Reply:
@jay-dee7 Sounds good.